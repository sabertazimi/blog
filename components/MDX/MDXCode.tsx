import { InlineCode } from '@/components/CodeBlocks'

interface Props {}

const MDXCode = (props: Props) => <InlineCode {...props} />

export default MDXCode
