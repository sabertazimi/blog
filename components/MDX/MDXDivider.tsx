import Divider from '@/components/Divider'

function MDXDivider(): JSX.Element {
  return <Divider className="bg-gradient-primary h-1 rounded-full" />
}

export default MDXDivider
