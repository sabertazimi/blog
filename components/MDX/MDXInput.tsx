import Checkbox from '@components/Checkbox'
import Input from '@components/Input'

interface Props {
  type?: string
  checked?: boolean
  disabled?: boolean
}

function MDXInput({
  type,
  checked,
  disabled,
  ...props
}: Props): JSX.Element {
  if (type?.includes('checkbox'))
    return <Checkbox checked={checked} className="mr-1" {...props} />

  return (
    <Input
      type={type}
      checked={checked}
      disabled={disabled}
      className="mr-1"
      {...props}
    />
  )
}

export default MDXInput
