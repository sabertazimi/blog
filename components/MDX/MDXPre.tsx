import { Pre } from '@components/CodeBlocks'

interface Props {}

const MDXPre = (props: Props) => <Pre {...props} />

export default MDXPre
