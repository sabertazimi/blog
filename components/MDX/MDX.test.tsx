import { render } from '@utils'
import MDXCode from './MDXCode'
import MDXDivider from './MDXDivider'
import MDXInput from './MDXInput'
import MDXPre from './MDXPre'

describe('MDXDivider', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<MDXDivider />)

    expect(container).toMatchSnapshot()
  })
})

describe('mDXInput', () => {
  it('should render input correctly (snapshot)', () => {
    const { container } = render(<MDXInput />)

    expect(container).toMatchSnapshot()
  })

  it('should render checkbox correctly (snapshot)', () => {
    const { container } = render(<MDXInput type="checkbox" />)

    expect(container).toMatchSnapshot()
  })
})

describe('mDXPre', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<MDXPre />)

    expect(container).toMatchSnapshot()
  })
})

describe('mDXCode', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<MDXCode />)

    expect(container).toMatchSnapshot()
  })
})
