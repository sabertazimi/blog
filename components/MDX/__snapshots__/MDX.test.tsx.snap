// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXCode should render correctly (snapshot) 1`] = `
<div>
  <code
    class="code dark:text-light dark:shadow-xl dark:shadow-primary"
  />
</div>
`;

exports[`MDXDivider should render correctly (snapshot) 1`] = `
<div>
  <div
    class="ant-divider ant-divider-horizontal divider bg-gradient-primary h-1 rounded-full dark:border-0 dark:!border-light dark:text-light"
    role="separator"
  />
</div>
`;

exports[`MDXInput should render checkbox correctly (snapshot) 1`] = `
<div>
  <label
    class="ant-checkbox-wrapper mr-1"
  >
    <span
      class="ant-checkbox"
    >
      <input
        class="ant-checkbox-input"
        type="checkbox"
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
  </label>
</div>
`;

exports[`MDXInput should render input correctly (snapshot) 1`] = `
<div>
  <input
    class="ant-input mr-1"
    type="text"
    value=""
  />
</div>
`;

exports[`MDXPre should render correctly (snapshot) 1`] = `
<div>
  <pre
    class="pre dark:shadow-xl dark:shadow-primary"
    data-language="TypeScript"
  >
    <button
      aria-label="CopyButton"
      class="outline-focus-visible absolute right-5 top-1.5 md:top-1"
      title="Copy"
      type="button"
    >
      <div
        class="wrapper"
        data-testid="toggle-wrapper"
        tabindex="-1"
      >
        <span
          aria-current="true"
          class="span"
          style="opacity: 1; transform: none;"
        >
          <span
            aria-label="copy"
            class="anticon anticon-copy icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="copy"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
              />
            </svg>
          </span>
        </span>
      </div>
    </button>
    <pre
      class="prism-code language-typescript code"
      style="color: rgb(118, 217, 230); background-color: rgb(42, 42, 42);"
    >
      <div
        class="token-line hover"
        style="color: rgb(118, 217, 230);"
      >
        <span
          class="number"
        >
          1
        </span>
        <span
          class="token plain"
          style="display: inline-block;"
        >
          

        </span>
      </div>
    </pre>
  </pre>
</div>
`;
